# Project X -- [Prisha] [Joshi]

**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [1]:
library(data.table)
options(repr.matrix.max.cols=50)
df <- fread("/anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales.csv", nrows= 1000)
df
df1<- fread("/anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales.csv", select = c(3:7))
df1

Invoice/Item Number Date       Store Number
1    S07122800016        08/16/2012 2846        
2    S21151400007        09/10/2014 3894        
3    S11572100060        04/10/2013 2558        
4    S07453200005        08/30/2012 4680        
5    S23684100152        01/26/2015 2590        
6    S06685900003        07/19/2012 4126        
7    S15834500013        11/20/2013 3417        
8    S15311500002        10/23/2013 4818        
9    S05332900031        05/02/2012 3403        
10   S10889600059        03/04/2013 3565        
11   S26804000069        07/16/2015 2595        
12   S25479500173        05/05/2015 2508        
13   S03636900001        01/18/2012 3645        
14   S12172100006        05/13/2013 4007        
15   S09751900060        12/27/2012 2633        
16   S10514100018        02/11/2013 4525        
17   S16590600035        12/30/2013 3964        
18   S03760800002        01/26/2012 2554        
19   S10214500010        01/24/2013 4622        
20   S04435900021        03/07/2012 3621        
21   S14609800010        09/17/2013 3676        
22   S22180700028        11/04/2014 5032        
23   S22157900025        11/04/2014 4637        
24   S20900100079        08/27/2014 2512        
25   S22645500005        12/01/2014 3790        
26   S05265300010        04/30/2012 4588        
27   S15646000074        11/11/2013 2603        
28   S25302800101        04/27/2015 2560        
29   S12662000013        06/10/2013 4676        
30   S04475300044        03/08/2012 2569        
...  ...                 ...        ...         
971  S06479900103        07/09/2012 2190        
972  S26411600068        06/25/2015 2579        
973  S04009000003        02/09/2012 4050        
974  S24775900264        03/30/2015 2573        
975  S12421900094        05/28/2013 2638        
976  S10759200125        02/25/2013 3692        
977  S03448200061        01/05/2012 2590        
978  S24714500091        03/25/2015 5105        
979  S23695900030        01/27/2015 2597        
980  S20222400102        07/21/2014 2633        
981  S26335400102        06/22/2015 4312        
982  S13518100016        07/23/2013 4373        
983  S07731500110        09/13/2012 4129        
984  S21422400011        09/24/2014 3990        
985  S04689700015        03/22/2012 4426        
986  S06289200023        06/27/2012 2611        
987  S04633400145        03/20/2012 3495        
988  S19353500030        06/04/2014 3964        
989  S23160800013        12/23/2014 5012        
990  S18575900017        04/23/2014 4183        
991  S25811900076        05/22/2015 2527        
992  S08104600002        10/04/2012 4393        
993  S09196000012        11/28/2012 3784        
994  S25153300021        04/20/2015 4148        
995  S11464400001        04/04/2013 4652        
996  S20932000054        08/28/2014 2595        
997  S14383300021        09/06/2013 4478        
998  S27729900072        09/04/2015 3612        
999  S13279900008        07/11/2013 4840        
1000 S20824500012        08/25/2014 4867        
     Store Name                           Address                   
1    CVS PHARMACY #8443 / CEDAR RAPIDS    3419 16TH AVE SW          
2    SMOKIN' JOE'S #6 TOBACCO AND LIQUOR  1404  1ST AVE NE          
3    HY-VEE FOOD STORE / MOUNT PLEASANT   1700 E WASHINGTON         
4    AFAL FOOD & LIQUOR / DES MOINES      4121 SE 14TH ST           
5    HY-VEE FOOD STORE #5 / CEDAR RAPIDS  3235 OAKLAND ROAD NE      
6    SAM'S MAINSTREET MARKET / SOLON      123 E MAIN ST             
7    BIG G FOOD STORE                     PO BOX 261  310 W DILLON  
8    DECORAH MART                         303, COLLEGE DR           
9    ECON-O-MART / COLUMBUS JUNCTION      113 EAST WALNUT ST.       
10   HARTIG DRUG STORE #10                701 MORMON TREK BLVD      
11   HY-VEE WINE AND SPIRITS / DENISON    1620  4TH AVE, SOUTH      
12   HY-VEE FOOD STORE #1 / CEDAR RAPIDS  1843 JOHNSON AVENUE, N.W. 
13   WAL-MART 1764 / WINDSOR HEIGHTS      1001  73RD ST             
1

Store Number Store Name                          
1        2846         CVS PHARMACY #8443 / CEDAR RAPIDS   
2        3894         SMOKIN' JOE'S #6 TOBACCO AND LIQUOR 
3        2558         HY-VEE FOOD STORE / MOUNT PLEASANT  
4        4680         AFAL FOOD & LIQUOR / DES MOINES     
5        2590         HY-VEE FOOD STORE #5 / CEDAR RAPIDS 
6        4126         SAM'S MAINSTREET MARKET / SOLON     
7        3417         BIG G FOOD STORE                    
8        4818         DECORAH MART                        
9        3403         ECON-O-MART / COLUMBUS JUNCTION     
10       3565         HARTIG DRUG STORE #10               
11       2595         HY-VEE WINE AND SPIRITS / DENISON   
12       2508         HY-VEE FOOD STORE #1 / CEDAR RAPIDS 
13       3645         WAL-MART 1764 / WINDSOR HEIGHTS     
14       4007         BROCKMANS                           
15       2633         HY-VEE #3 / BDI / DES MOINES        
16       4525         CASEY'S GENERAL STORE #2425 / AKRON 
17       3964         SIOUX FOOD CENTER OF SIOUX RAPIDS   
18       2554         HY-VEE FOOD STORE / DAVENPORT       
19       4622         QUIK TRIP #544 / SE 14TH DM         
20       3621         JENSEN LIQUORS, LTD.                
21       3676         T AND M FOODS                       
22       5032         FASTMART                            
23       4637         KWIK SHOP #560 / CEDAR RAPIDS       
24       2512         HY-VEE WINE AND SPIRITS / IOWA CITY 
25       3790         WAL-MART 0784 / MT PLEASAN          
26       4588         SAM'S MINI MART / SIOUX CITY        
27       2603         HY-VEE WINE AND SPIRITS / BETTENDORF
28       2560         HY-VEE FOOD STORE / MARION          
29       4676         SAM'S MINI MART / SIOUX CITY        
30       2569         HY-VEE FOOD STORE #2 / CEDAR RAPIDS 
...      ...          ...                                 
27050114 2552         HY-VEE FOOD STORE #3 / CEDAR RAPIDS 
27050115 3890         SMOKIN' JOE'S #7 TOBACCO AND LIQUOR 
27050116 4433         KUM & GO #24  PLEASANT HILL         
27050117 3612         B AND C LIQUOR / MAQUOKETA          
27050118 4829         CENTRAL CITY 2                      
27050119 4437         KUM & GO #215  AMES                 
27050120 4315         FAREWAY STORES #554 / WASHINGTON    
27050121 2600         HY-VEE FOOD STORE / OSKALOOSA       
27050122 4898         BURLINGTON SHELL                    
27050123 4222         STATION MART LIQUOR AND TOBBACCO    
27050124 4231         FAREWAY STORES #719 / LE MARS       
27050125 2248         INGERSOLL LIQUOR AND BEVERAGE       
27050126 2634         HY-VEE WINE AND SPIRITS / AMES      
27050127 4471         KUM & GO #212 / WATERLOO            
27050128 2625         HY-VEE WINE AND SPIRITS #2          
27050129 3818         ROUND WINDOW LIQUOR                 
27050130 3697         OKY DOKY #2 / DUBUQUE               
27050131 4170         DEWEY'S JACK AND JILL               
27050132 2539         HY-VEE FOOD STORE / IOWA FALLS      
27050133 2650         HY-VEE WINE AND SPIRITS / HARLAN    
27050134 4467         KUM & GO #438 / MUSCATINE           
27050135 4108         OASIS GROCERY AND LIQUOR /          
27050136 2513         HY-VEE FOOD STORE #2 / IOWA CITY    
27050137 2448         BOTTLE BOUTIQUE                     
27050138 2587         HY-VEE FOOD STORE / JOHNSTON        
27050139 3853         RAM LIQUOR MART                     
27050140 4092         FAREWAY STORES #077 / NORWALK       
27050141 2544         HY-VEE FOOD STORE / MARSHALLTOWN    
27050142 2567         HY-VEE DRUGSTORE / DAVENPORT        
27050143 2558         HY-VEE FOOD STORE / MOUNT PLEASANT  
         Address                       City              Zip Code
1        3419 16TH AVE SW              CEDAR RAPIDS      52404   
2        1404  1ST AVE NE              CEDAR RAPIDS      52402   
3        1700 E WASHINGTON             MOUNT PLEASANT    52641   
4        4121 SE 14TH ST               DES MOINES        50320   
5        3235 OAKLAND

Columns 3,4,5,6,and 7 have contain the Store number , name, address, city and zip code respectively. the dimension is 5 * 27050143 

## Question 2

In [2]:
library(data.table)
# ten most popular store numbers
df1 <- fread("/anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales.csv", select = c(3:7))
store_number <- sort(table(df1$'Store Number'),decreasing = TRUE)
head(store_number , 10)
# ten most popular store names 
store_name <- sort(table(df1$'Store Name'),decreasing = TRUE)
head(store_name ,10)
#zip, address and city pasted together 
acz_ten <- sort(table(paste(df1$Address , df1$City , df1$'Zip Code')))
head(acz_ten , 10)


  2633   4829   2190   2512   2572   2603   2515   2614   2647   2648 
223447 190702 173109 147581 142536 140319 133959 132915 132269 128608 


        HY-VEE #3 / BDI / DES MOINES                       CENTRAL CITY 2 
                              223447                               190653 
           CENTRAL CITY LIQUOR, INC.      HY-VEE FOOD STORE / CEDAR FALLS 
                              173158                               142536 
 HY-VEE WINE AND SPIRITS / IOWA CITY HY-VEE WINE AND SPIRITS / BETTENDORF 
                              141557                               140319 
            HY-VEE #7 / CEDAR RAPIDS                      HY-VEE #4 / WDM 
                              131294                               128608 
   HY-VEE FOOD STORE #1 / MASON CITY                    BENZ DISTRIBUTING 
                              127264                               120357 


                                DUBUQUE 52002 
                                            1 
             29538, 210TH ST. KEOSAUQUA 52565 
                                            1 
                 3345 270TH AVE. KEOKUK 52632 
                                            1 
1685 W UINTAH ST. #101 COLORADO SPRINGS 80904 
                                            2 
       213 S 3RD ST. PO BOX 106 RAYMOND 50667 
                                            2 
             3576, DOLPHIN DR IOWA CITY 52240 
                                            2 
       4860 HEATHERSTONE RD. BETTENDORF 52722 
                                            2 
                  314 S 5TH ST CHEROKEE 51012 
                                            5 
                318 E 2ND ST. DAVENPORT 52801 
                                            5 
               123 MAIN ST #B TEMPLETON 51463 
                                            6 

no, the answers in paqrt a and b dont exactly agree and neither do they agree in part a and c. 

## Question 3

In [12]:
library(data.table)
df1 <- fread("/anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales.csv", select = c(7,12,22))
code_sales <- tapply(df1$'Sale (Dollars)', df1$'Zip Code', sum)
top_ten <- sort(code_sales, decreasing = TRUE )
print(head(top_ten , 10 ))
name_sales <- tapply( df1$'Sale (Dollars)',df1$'Category Name', sum)
top_ten_category <- sort(name_sales , decreasing = TRUE)
print(head(top_ten_category , 10 ))

    50320     52402     52240     50266     51501     52241     50613     50311 
132861227 108460935 106827909  95956449  84485599  80224356  70716357  65407917 
    52722     50021 
 63447651  61328202 
        CANADIAN WHISKIES           AMERICAN VODKAS STRAIGHT BOURBON WHISKIES 
                457612891                 380307151                 257794862 
               SPICED RUM           WHISKEY LIQUEUR           IMPORTED VODKAS 
                254362805                 199736755                 183082359 
       TENNESSEE WHISKIES        100% AGAVE TEQUILA          BLENDED WHISKIES 
                162676709                 124223944                 109152591 
        IMPORTED BRANDIES 
                 88413646 


Markdown notes and sentences and analysis written here.

## Question 4

In [1]:
library(data.table)
df2 <- fread("/anvil/projects/tdm/data/election/itcont2020.txt" , select = c (10,15))
head(df2)
state_sales <- tapply(df2$V15, df2$V10 , sum )
top_ten_sales <- sort(state_sales , decreasing = TRUE)
print(head(top_ten_sales , 10 ))

Warning message in fread("/anvil/projects/tdm/data/election/itcont2020.txt", select = c(10, :
"Found and resolved improper quoting out-of-sample. First healed line 543881: <<C00401224|A|M10||202012149381299609|24T|IND|YOUNG, BRUCE|SARANAC LAKE|NY|12983|"SELF" (EMPLOYER WISHES TO REMAIN ANON|IND. CONTRACTOR / CARETAKER|09292020|27|C00649061|SA11AI_310755484|1482832||EARMARKED FOR COMMITTEE TO ELECT TEDRA COBB (C00649061)|4012120212130550457>>. If the fields are not quoted (e.g. field separator does not appear within any field), try quote="" to avoid this warning."


V10 V15
1 TX   25
2 MO  100
3 FL    5
4 FL    5
5 FL   50
6 CA  100

        CA         NY         VA         DC         TX         FL         IL 
2803518256 2521628477 1122819081 1060820708 1005943807  913762179  610913249 
        MA         NV         WA 
 521866708  384867853  384169499 


Markdown notes and sentences and analysis written here.

## Question 5

In [ ]:
library(data.table)
data <- fread("/anvil/projects/tdm/data/movies_and_tv/imdb2024/basics.tsv" )
head(data , nrows = 10 )
data$genres

strsplit(data$genres, ',')

unlist(strsplit(data$genres, ','))

table(unlist(strsplit(data$genres, ',')))


Warning message in fread("/anvil/projects/tdm/data/movies_and_tv/imdb2024/basics.tsv"):
"Found and resolved improper quoting out-of-sample. First healed line 242860: <<tt0253684	tvEpisode	"Rome br<c3><bb>le" (Portrait de Shirley Clarke)	"Rome br<c3><bb>le" (Portrait de Shirley Clarke)	0	1970		54	Documentary>>. If the fields are not quoted (e.g. field separator does not appear within any field), try quote="" to avoid this warning."


tconst    titleType primaryTitle           originalTitle          isAdult
1 tt0000001 short     Carmencita             Carmencita             0      
2 tt0000002 short     Le clown et ses chiens Le clown et ses chiens 0      
3 tt0000003 short     Poor Pierrot           Pauvre Pierrot         0      
4 tt0000004 short     Un bon bock            Un bon bock            0      
5 tt0000005 short     Blacksmith Scene       Blacksmith Scene       0      
6 tt0000006 short     Chinese Opium Den      Chinese Opium Den      0      
  startYear endYear runtimeMinutes genres                  
1 1894      NA       1             Documentary,Short       
2 1892      NA       5             Animation,Short         
3 1892      NA       5             Animation,Comedy,Romance
4 1892      NA      12             Animation,Short         
5 1893      NA       1             Comedy,Short            
6 1894      NA       1             Short

In [1]:
library(data.table)
data2 <- fread("/anvil/projects/tdm/data/movies_and_tv/imdb2024/basics.tsv", select = "genres")
head(data2$genres)
all_genres <- unlist(strsplit(data2$genres, ','))
genre_counts <- table(all_genres)
sorted_genre_counts <- sort(genre_counts, decreasing = TRUE)
head(sorted_genre_counts,10)

Warning message in fread("/anvil/projects/tdm/data/movies_and_tv/imdb2024/basics.tsv", :
"Found and resolved improper quoting out-of-sample. First healed line 242860: <<tt0253684	tvEpisode	"Rome br<c3><bb>le" (Portrait de Shirley Clarke)	"Rome br<c3><bb>le" (Portrait de Shirley Clarke)	0	1970		54	Documentary>>. If the fields are not quoted (e.g. field separator does not appear within any field), try quote="" to avoid this warning."


[1] "Documentary,Short"        "Animation,Short"         
[3] "Animation,Comedy,Romance" "Animation,Short"         
[5] "Comedy,Short"             "Short"

all_genres
      Drama      Comedy   Talk-Show       Short Documentary        News 
    3151064     2181847     1372500     1191319     1062294     1051399 
    Romance      Family  Reality-TV   Animation 
    1045327      824607      624854      556566 

Markdown notes and sentences and analysis written here.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.